cardano-cli/src/Cardano/CLI/EraBased/Transaction/Command.hs

```diff
@@ -245,11 +250,48 @@ data TransactionCalculateMinValueCmdArgs era = TransactionCalculateMinValueCmdAr
   deriving Show
 
 data TransactionCalculatePlutusScriptCostCmdArgs = TransactionCalculatePlutusScriptCostCmdArgs
-  { nodeConnInfo :: !LocalNodeConnectInfo
+  { nodeContextInfoSource :: !NodeContextInfoSource
   , txFileIn :: FilePath
   , outputFile :: !(Maybe (File () Out))
   }
 
+-- | Either information about the context in which the transaction command
+-- is run, or information required to obtain it (information to connect to the node).
+data NodeContextInfoSource
+  = NodeConnectionInfo !LocalNodeConnectInfo
+  | ProvidedTransactionContextInfo !TransactionContext
+
+-- | Transaction context, requried to evaluate the execution
+-- costs of the plutus scripts in the transaction.
+data TransactionContext = TransactionContext
+  { systemStartSource :: SystemStartOrGenesisFileSource
+  , mustExtendSafeZone :: MustExtendSafeZone
+  , eraHistoryFile :: File EraHistory In
+  , utxoFile :: File () In
+  , protocolParamsFile :: ProtocolParamsFile
+  }
+
+-- | The system start time or a means to get it (the genesis file)
```

Comment:
"The system start time or the genesis file from which to get it"

Reply:
Addressed here: https://github.com/IntersectMBO/cardano-cli/commit/b3bc70dea9db4ec1eba733a79586a8f0b0ce55e3